### DISCUSSION TITLE
Document using code generation with unity

### DISCUSSION THREAD

**Discussion by qleguennec** [louthy/language-ext#1285] [Q&A]
I have this in a unity script:

```
    [Union]
    public interface IProperty
    {
        IProperty Transform(float x, float y, float z);
        IProperty Color(float r, float g, float b);
        IProperty Name(string name);
    }
```

But I'm not getting any completion with `IProperty.Color`. I'm guessing it's because code generation isn't generating anything.
Does it have anything to do with unity? If so, how to solve it?

**Comment by louthy** [comment]
I think it's unlikely that Unity supports it, but I don't use Unity so I've never tested it.  

If anybody else has had success running the codegen on Unity please let us know your setup.

**Comment by qleguennec** [comment]
Is there an easy setup where I can generate code in eg visual studio and then use it in unity ?

**Comment by louthy** [comment]
In theory, yes, as the code-gen just creates generated `.cs` files that are then compiled as normal.  For example, the `TestBed` project in the `Samples` folder has some code-gen'd types, and it produces the file shown below (in yellow):

![image](https://github.com/louthy/language-ext/assets/1991827/5c263354-04a8-43ff-adb8-17d0c7560205)
